### PR TITLE
[x/programs] add runtime client

### PR DESCRIPTION
**Pull request by hexfusion**
This PR adds a client abstraction around the wazero api.Module for managing exported functions of the guest. This PR also adds memory management for any allocations made during the client life cycle.

### CODE REVIEW

File: x/programs/examples/token.go [hunk 1]
```diff
@@ -112,22 +112,6 @@ func (t *Token) Run(ctx context.Context) error {
 		zap.Int64("alice", int64(result[0])),
 	)
 
-	// deallocate bytes
```
Comment (hexfusion) [inline x/programs/examples/token.go:115]: this is handled automatically by the client

File: x/programs/examples/token.go [hunk 2]
```diff
@@ -40,8 +40,6 @@ func (t *Token) Run(ctx context.Context) error {
 		"mint_to",
 		"get_balance",
 		"transfer",
-		"alloc",
-		"dealloc",
```
Comment (hexfusion) [inline x/programs/examples/token.go:44]: I think we can assume these will always be supported so the user should not need to define them. please conform @samliok

Comment (samliok) [reply x/programs/examples/token.go:44]: yea 👍

**Review by richardpringle — DISMISSED** [review]
Not that my approval means much 😉

**Pull request closed without merging**